vault/src/main.rs

```diff
@@ -137,4 +149,32 @@ mod tests {
             task.await.unwrap().unwrap();
         }
     }
+
+    #[tokio::test]
+    async fn test_vault_pid_file() {
+        let dummy_account_id = AccountId::new(Default::default());
+        let dummy_spec_name = "kintsugi-testnet".to_string();
+        let termination_signals = &[SIGHUP, SIGTERM, SIGINT, SIGQUIT];
+        let mut sys = System::new_all();
+
+        let task = tokio::spawn({
+            let _pidfile = PidFile::create(&dummy_spec_name, &dummy_account_id, &mut sys).unwrap();
```

Comment:
Note that this file cannot pollute the filesystem on a test crash. Pid files are [created](https://github.com/savudani8/interbtc-clients/blob/c87ccfb824720f22eb6e82820a7121c0d8977dac/vault/src/process.rs#L80) in `std::env::temp_dir()`, which is also what `TempDir` [uses](https://docs.rs/tempdir/latest/src/tempdir/lib.rs.html#159-161). It also implements `Drop`, which is still called on crash (`TempDir` relies on this assumption as well).